### ISSUE TITLE
Consider making _GetProjectReferenceTargetFrameworkProperties target incremental

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#2309]
Currently when building a codebase the second time the only target we see built from scratch is:
`Building target "_GetProjectReferenceTargetFrameworkProperties" completely.`

I realize that it's possibly by design but it would be really convenient if that target somehow specified inputs and outputs if at all possible. This way we can simply search the build log for "Building target completely" and if we didn't find it, it means the build was fully incremental with no unnecessary rebuilding.

This is the only target that I see currently always "rebuilding", even though it may be by design.

This is a nice to have.

**Comment by KirillOsenkov** [comment]
![image](https://user-images.githubusercontent.com/679326/28249821-224d8d70-6a12-11e7-9076-17e6a1d4a4a0.png)

**Comment by rainersigwald** [comment]
This will be fixed with basically any fix of the double-evaluation problem. The reason there's an `Output` specified for that target is just to batch over multiple references, which should be fixed to do them all in parallel (and my proposed improvements do just that).

Because of that I'm going to claim this is a duplicate of #1276 even though it could be fixed independently (by switching the batching to be based on an always-true condition).

**Comment by KirillOsenkov** [comment]
Perfect, that works for me!

**Comment by rainersigwald** [comment]
And this is actually fully a duplicate of #1785, so I'm going to close it.
